src/Controller/Component/SecurityComponent.php

```diff
@@ -104,7 +104,7 @@ public function startup(Event $event)
         $this->_secureRequired($controller);
         $this->_authRequired($controller);
 
-        $isPost = $this->request->is(['post', 'put']);
+        $hasData = !empty($this->request->data);
```

Comment:
Could only now take a look.
Thanks for doing this "op-in" instead of "op-out" security principle <3.